### PR TITLE
Es plugin fix

### PR DESCRIPTION
**Pull request by ansakharov**
Fix parse_es, so process no longer crushes on invalid elastic search query, it's goes to discard now.
Also add some refactoring dictated by golangci-lint.

### CODE REVIEW

**Review by anuriq — APPROVED** [review]
LGTM

File: plugin/output/file/file.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ func (p *Plugin) createNew() {
 	logger.Errorf("tsFileName in createNew=%s", p.tsFileName)
 	f := fmt.Sprintf("%s%s", p.targetDir, p.tsFileName)
 	pattern := fmt.Sprintf("%s*%s%s%s", p.targetDir, fileNameSeparator, p.fileName, p.fileExtension)
-	matches, err := filepath.Glob(pattern)
```
Comment (vano144) [inline plugin/output/file/file.go:209]: maybe here, we can handle error?

File: plugin/input/http/README.md [hunk 2]
```diff
@@ -22,5 +22,39 @@ Which protocol to emulate.
...
+        brokers: [kafka-local:9092, kafka-local:9091]
+        default_topic: yourtopic-k8s-data
+        use_topic_field: true
+        topic_field: pipeline_kafka_topic
+      
+      # Or we can write to file: 
+      # type: file
+      # target_file: "./output.txt"
+```
+
```
Comment (vano144) [inline plugin/input/http/README.md:58]: A couple of test queries also be helpful here

File: plugin/input/file/file_test.go [hunk 3]
```diff
@@ -244,7 +244,7 @@ func addLines(file string, from int, to int) int {
 	size := 0
 	for i := from; i < to; i++ {
 
-		str := fmt.Sprintf(`"_`)
+		str := `"_`
```
Comment (vano144) [inline plugin/input/file/file_test.go:247]: maybe constant here?

File: plugin/action/parse_es/parse_es.go [hunk 4]
```diff
@@ -80,7 +79,6 @@ func (p *Plugin) Do(event *pipeline.Event) pipeline.ActionResult {
 		return pipeline.ActionCollapse
 	}
 
-	p.logger.Fatalf("wrong ES input format, expected action, got: %s", root.EncodeToString())
-
+	// If request invalid skip bad event.
```
Comment (vano144) [inline plugin/action/parse_es/parse_es.go:82]: maybe add test on it?

File: plugin/input/file/file_test.go (hunk 3 above)
Comment (ansakharov) [reply plugin/input/file/file_test.go]: Added

File: plugin/action/parse_es/parse_es.go (hunk 4 above)
Comment (ansakharov) [reply plugin/action/parse_es/parse_es.go:82]: I wrote units on it.

File: plugin/output/file/file.go (hunk 1 above)
Comment (ansakharov) [reply plugin/output/file/file.go:209]: I'm not sure for now, that err must be thrown, so err skipped as it was.

File: plugin/input/http/README.md (hunk 2 above)
Comment (ansakharov) [reply plugin/input/http/README.md:58]: I added example

**Review by vano144 — APPROVED** [review]

File: plugin/action/parse_es/parse_es.go (hunk 4 above)
Comment (vitkovskii) [reply plugin/action/parse_es/parse_es.go:82]: Maybe we should log an error here?

File: plugin/action/parse_es/parse_es_test.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+			})
+
+			for _, event := range tCase.eventsIn {
+				input.In(event.sourceID, event.sourceName, event.offset, event.bytes)
+			}
+
+			wg.Wait()
+			p.Stop()
+
+			assert.Equal(t, 2, inEvents, "wrong eventsIn events count")
```
Comment (vitkovskii) [inline plugin/action/parse_es/parse_es_test.go:126]: Fix typo here

File: plugin/action/parse_es/parse_es_test.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+					bytes:      []byte(invalidMsg),
+				},
+				{
+					sourceID:   pipeline.SourceID(1),
+					sourceName: "parse_es_test_source.log",
+					offset:     0,
+					bytes:      []byte(invalidMsg),
+				},
+			},
+			incIn:  2,
```
Comment (vitkovskii) [inline plugin/action/parse_es/parse_es_test.go:49]: `eventsIn`/`eventsOut`

File: plugin/action/parse_es/parse_es_unit_test.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
+package parse_es
```
Comment (vitkovskii) [inline plugin/action/parse_es/parse_es_unit_test.go:1]: It's a default that test is a unit test. Let's rename another file to `pipeline_test`?

File: plugin/output/file/file.go (hunk 1 above)
Comment (vitkovskii) [reply plugin/output/file/file.go:209]: if we can't run glob, I think we should do a fatal

Comment (ansakharov) [reply plugin/output/file/file.go:209]: Oh, I thought that we can just log and ignore it. Added log.Fatalf

File: plugin/action/parse_es/parse_es.go (hunk 4 above)
Comment (ansakharov) [reply plugin/action/parse_es/parse_es.go:82]: Goddamright. Added log and test.

File: plugin/action/parse_es/parse_es_unit_test.go (hunk 7 above)
Comment (ansakharov) [reply plugin/action/parse_es/parse_es_unit_test.go]: `pipeline_test` will intersect with other `pipeline_test` tests, is it ok? 
That's probably my issue - I used to name `package_name_test` - for units, `package_name_integration_test` - for integration.

File: plugin/output/file/file.go [hunk 8]
```diff
@@ -207,6 +207,9 @@ func (p *Plugin) createNew() {
 	f := fmt.Sprintf("%s%s", p.targetDir, p.tsFileName)
 	pattern := fmt.Sprintf("%s*%s%s%s", p.targetDir, fileNameSeparator, p.fileName, p.fileExtension)
 	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		p.logger.Fatalf("can't bad search pattern: %s, err: %s", pattern, err.Error())
```
Comment (vitkovskii) [inline plugin/output/file/file.go:211]: `can't glob: pattern=%s, err=%s`

**Comment by ansakharov** [comment]
@vitkovskii I think we can merge this.

**Pull request merged**